app/src/main/java/com/gdgcatania/kotlinbootcamp/RetrofitService.kt

```diff
@@ -0,0 +1,49 @@
+package com.gdgcatania.kotlinbootcamp
+
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+/**
+ * Retrofit Service builder used for getting enabling the use of the WheaterService
+ *
+ * Kotlin project for the #DevFestMed bootcamp
+ * Created by carlo on 19/11/2017.
+ */
+
+class RetrofitService {
+
+    /**
+     * return a @see WheaterService instance
+     */
+    fun createWeatherService(): WeatherService {
+
+        // interceptor for custom request (add API_KEY and Custom settings as query)
+
+        val interceptor = object : Interceptor{
+            override fun intercept(chain: Interceptor.Chain?): Response? {
```

Comment:
In generale, controlla sempre le annotation dei parametri: In Kotlin, quando arrivi da Java, ti dice che una cosa è `nullable` di default. Se guardi il parametro di `intercept`, non è annotato con `@Nullable`, possiamo dare per scontato che quindi sia non-nullo. Prova a rimuovere il `?` da `Interceptor.Chain`